### ISSUE TITLE
ApplicationIcon does not work when cross building win-x64 from Linux

### ISSUE THREAD

**Issue by NicolasDorier** [dotnet/sdk#3943]
# How to reproduce

On linux (using Buster 3.0.101):
```bash
dotnet new console
# Add myicon.ico to the project
dotnet publish --runtime win-x64 -p:ApplicationIcon=myicon.ico
```
Grab `bin/Debug/netcoreapp3.0/publish/*.exe`
Notice that the executable does not show any icon on windows explorer.
The same process works properly show the icon on windows

# Expected

The executable should have the application icon properly set.

# Version

```
.NET Core SDK (reflecting any global.json):
 Version:   3.0.101
 Commit:    bc5f8df0f5

Runtime Environment:
 OS Name:     debian
 OS Version:  10
 OS Platform: Linux
 RID:         debian.10-x64
 Base Path:   /usr/share/dotnet/sdk/3.0.101/

Host (useful for support):
  Version: 3.0.1
  Commit:  32085cbc72

.NET Core SDKs installed:
  3.0.101 [/usr/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.0.1 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.0.1 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by frankracis-work** [comment]
It's not limited to icons.   Version information isn't present, either.

No PE resources at all are added to the Win32 EXE file when published on Linux.

Using the "peres" utility from "pev", If built on Linux:
```peres -a bin/release/netcoreapp3.1/win-x64/publish/hello.exe```
returns 
```this file has no resources```
If built on Windows, the same command will list resources and dump values..

Still present on the latest build: SDK 3.1.200, Host 3.1.2/916b5cba26 on both Linux and Windows.

**Comment by KoalaBear84** [comment]
Yes. I found out because I switched to ubuntu for building new releases with GitHub Actions, so switched back to Windows.

Hopefully will be fixed. Building for Windows should remain on Windows unless this is fixed.

**Comment by augustocb23** [comment]
Still not working on `net5.0`. Tested on Ubuntu and Docker (Dotnet SDK image)

**Comment by KoalaBear84** [comment]
It would be nice if someone from Microsoft could respond to this about what the reason is and when it will be fixed.

**Comment by darkguy2008** [comment]
+1, I just found out this through a Google search because my app built on the cloud in Docker wasn't having an app icon. Is there any timeline from the MS team to fix this?

**Comment by eduherminio** [comment]
I've identified the same problem cross building win-x64 from macOS (`macOS-latest` GH runner).
So... no other option that building from Windows afaik.

**Comment by NicolasDorier** [comment]
@eduherminio there is solutions, look at my workaround with wine and rcedit.

**Comment by KalleOlaviNiemitalo** [comment]
May be a duplicate of <https://github.com/dotnet/runtime/issues/3828>.

**Comment by NicolasDorier** [comment]
How to say it will never be fixed without telling it will never be fixed

![image](https://user-images.githubusercontent.com/3020646/143857329-a46f84f1-0fe4-4637-b08d-75d68a8b4dcf.png)

**Comment by keithfin** [comment]
For anyone else that landed on this thread, confirming 6.0.1 is still affected

**Comment by MaverickMartyn** [comment]
@NicolasDorier According to [dotnet/runtime#3828](https://github.com/dotnet/runtime/issues/3828) this is scheduled for 7.0.0 again.
No need to assume anything. It was likely just moved to "Future" temporarily as it was being investigated.

**Comment by markusschaber** [comment]
> @NicolasDorier According to [dotnet/runtime#3828](https://github.com/dotnet/runtime/issues/3828) this is scheduled for 7.0.0 again. No need to assume anything. It was likely just moved to "Future" temporarily as it was being investigated. :)

I can confirm that 7.0 is still affected. :-(

**Comment by NicolasDorier** [comment]
I believe 8.0 now fixed it and this can be closed (https://github.com/dotnet/core/issues/8439#issuecomment-1688474268)

**Comment by downthecrop** [comment]
> I believe 8.0 now fixed it and this can be closed ([dotnet/core#8439 (comment)](https://github.com/dotnet/core/issues/8439#issuecomment-1688474268)), didn't try yet though.

Was able to test my pipeline with `mcr.microsoft.com/dotnet/sdk:8.0` RC. Successfully cross compiled with the ico! 💯

**Comment by markusschaber** [comment]
Seems to work on my side with .NET 8 SDK, the fix might have been https://github.com/dotnet/runtime/issues/3828.
